From f7e2fcd702a95fd001140041dce23d6c5f1339bc Mon Sep 17 00:00:00 2001
From: Michael W <baddaemon87@gmail.com>
Date: Sat, 6 Jan 2018 13:49:46 +0100
Subject: [PATCH 060/249] wiki: Add validation for cpu

* Commonize the format
* Add a list of valid cpus
* Correct a few

Change-Id: I734525da710093b3f8b60e3b27d391ed842cc2d6
---
 _data/devices/A6020.yml        |  2 +-
 _data/devices/P024.yml         |  2 +-
 _data/devices/Z008.yml         |  4 ++--
 _data/devices/Z00A.yml         |  4 ++--
 _data/devices/Z00D.yml         |  2 +-
 _data/devices/Z00L.yml         |  2 +-
 _data/devices/Z00T.yml         |  2 +-
 _data/devices/addison.yml      |  2 +-
 _data/devices/angler.yml       |  2 +-
 _data/devices/armani.yml       |  2 +-
 _data/devices/athene.yml       |  2 +-
 _data/devices/chagallwifi.yml  |  2 +-
 _data/devices/che10.yml        |  2 +-
 _data/devices/cherry.yml       |  2 +-
 _data/devices/condor.yml       |  2 +-
 _data/devices/crackling.yml    |  2 +-
 _data/devices/d800.yml         |  2 +-
 _data/devices/d801.yml         |  2 +-
 _data/devices/d802.yml         |  2 +-
 _data/devices/d803.yml         |  2 +-
 _data/devices/espresso3g.yml   |  2 +-
 _data/devices/espressowifi.yml |  2 +-
 _data/devices/falcon.yml       |  2 +-
 _data/devices/flounder.yml     |  2 +-
 _data/devices/flounder_lte.yml |  2 +-
 _data/devices/foster.yml       |  2 +-
 _data/devices/g2m.yml          |  2 +-
 _data/devices/gohan.yml        |  2 +-
 _data/devices/gtelwifiue.yml   |  2 +-
 _data/devices/gtesqltespr.yml  |  2 +-
 _data/devices/gts210ltexx.yml  |  2 +-
 _data/devices/gts210vewifi.yml |  2 +-
 _data/devices/gts210wifi.yml   |  2 +-
 _data/devices/gts28vewifi.yml  |  2 +-
 _data/devices/ha3g.yml         |  2 +-
 _data/devices/hammerhead.yml   |  2 +-
 _data/devices/harpia.yml       |  2 +-
 _data/devices/hero2lte.yml     |  2 +-
 _data/devices/herolte.yml      |  2 +-
 _data/devices/hiaeuhl.yml      |  2 +-
 _data/devices/hiaeul.yml       |  2 +-
 _data/devices/himaul.yml       |  2 +-
 _data/devices/himawl.yml       |  2 +-
 _data/devices/hydrogen.yml     |  2 +-
 _data/devices/i9100.yml        |  2 +-
 _data/devices/i9300.yml        |  2 +-
 _data/devices/i9305.yml        |  2 +-
 _data/devices/ido.yml          |  2 +-
 _data/devices/ivy.yml          |  2 +-
 _data/devices/jag3gds.yml      |  2 +-
 _data/devices/jagnm.yml        |  2 +-
 _data/devices/jalebi.yml       |  2 +-
 _data/devices/k3gxx.yml        |  2 +-
 _data/devices/karin.yml        |  2 +-
 _data/devices/karin_windy.yml  |  2 +-
 _data/devices/kenzo.yml        |  2 +-
 _data/devices/kipper.yml       |  2 +-
 _data/devices/klimtwifi.yml    |  2 +-
 _data/devices/lettuce.yml      |  2 +-
 _data/devices/lux.yml          |  2 +-
 _data/devices/m216.yml         |  2 +-
 _data/devices/maguro.yml       |  2 +-
 _data/devices/merlin.yml       |  2 +-
 _data/devices/mido.yml         |  2 +-
 _data/devices/mondrianwifi.yml |  2 +-
 _data/devices/mt2.yml          |  2 +-
 _data/devices/n1awifi.yml      |  2 +-
 _data/devices/n2awifi.yml      |  2 +-
 _data/devices/n5100.yml        |  2 +-
 _data/devices/n5110.yml        |  2 +-
 _data/devices/n5120.yml        |  2 +-
 _data/devices/nx512j.yml       |  2 +-
 _data/devices/oneplus2.yml     |  2 +-
 _data/devices/osprey.yml       |  2 +-
 _data/devices/otus.yml         |  2 +-
 _data/devices/paella.yml       |  2 +-
 _data/devices/peach.yml        |  2 +-
 _data/devices/peregrine.yml    |  2 +-
 _data/devices/piccolo.yml      |  2 +-
 _data/devices/r5.yml           |  2 +-
 _data/devices/r7plus.yml       |  2 +-
 _data/devices/r7sf.yml         |  2 +-
 _data/devices/roth.yml         |  2 +-
 _data/devices/seed.yml         |  2 +-
 _data/devices/shieldtablet.yml |  2 +-
 _data/devices/surnia.yml       |  2 +-
 _data/devices/suzuran.yml      |  2 +-
 _data/devices/t0lte.yml        |  2 +-
 _data/devices/tenshi.yml       |  2 +-
 _data/devices/thea.yml         |  2 +-
 _data/devices/titan.yml        |  2 +-
 _data/devices/tomato.yml       |  2 +-
 _data/devices/toro.yml         |  2 +-
 _data/devices/toroplus.yml     |  2 +-
 _data/devices/v1awifi.yml      |  2 +-
 _data/devices/v400.yml         |  2 +-
 _data/devices/v410.yml         |  2 +-
 _data/devices/v480.yml         |  2 +-
 _data/devices/v500.yml         |  2 +-
 _data/devices/v521.yml         |  2 +-
 _data/devices/vegetalte.yml    |  2 +-
 _data/devices/w5.yml           |  2 +-
 _data/devices/w7.yml           |  2 +-
 _data/devices/wt88047.yml      |  2 +-
 test/schema-06.yml             | 23 +++++++++++++++++++++++
 105 files changed, 129 insertions(+), 106 deletions(-)

diff --git a/_data/devices/A6020.yml b/_data/devices/A6020.yml
index b5771be..a283818 100644
--- a/_data/devices/A6020.yml
+++ b/_data/devices/A6020.yml
@@ -7,7 +7,7 @@ cameras:
 carrier:
 channels: [weekly]
 codename: A6020
-cpu: Cortex A53 & A53
+cpu: Cortex-A53 & Cortex-A53
 cpu_cores: '8'
 cpu_freq: 4 x 1.5 GHz & 4 x 1.2 GHz
 current_branch: 14.1
diff --git a/_data/devices/P024.yml b/_data/devices/P024.yml
index b106633..0d44ae5 100644
--- a/_data/devices/P024.yml
+++ b/_data/devices/P024.yml
@@ -7,7 +7,7 @@ cameras:
 carrier:
 channels: [weekly]
 codename: P024
-cpu: Cortex A53
+cpu: Cortex-A53
 cpu_cores: '4'
 cpu_freq: 1.2 GHz
 current_branch: 14.1
diff --git a/_data/devices/Z008.yml b/_data/devices/Z008.yml
index 9fea2dd..311488e 100644
--- a/_data/devices/Z008.yml
+++ b/_data/devices/Z008.yml
@@ -1,4 +1,4 @@
-architecture: x86
+architecture: x86_64
 battery: 3000mAh
 bluetooth: {spec: '4.0', profiles: [A2DP]}
 cameras:
@@ -7,7 +7,7 @@ cameras:
 carrier:
 channels: [weekly]
 codename: Z008
-cpu: x86_64
+cpu: Intel Atom
 cpu_cores: '4'
 cpu_freq: 1.8 GHz
 current_branch: 14.1
diff --git a/_data/devices/Z00A.yml b/_data/devices/Z00A.yml
index 566a2a5..33a6c22 100644
--- a/_data/devices/Z00A.yml
+++ b/_data/devices/Z00A.yml
@@ -1,4 +1,4 @@
-architecture: x86
+architecture: x86_64
 battery: 3000mAh
 bluetooth: {spec: '4.0', profiles: [A2DP]}
 cameras:
@@ -7,7 +7,7 @@ cameras:
 carrier:
 channels: [weekly]
 codename: Z00A
-cpu: x86_64
+cpu: Intel Atom
 cpu_cores: '4'
 cpu_freq: 2.3/1.8 GHz
 current_branch: 14.1
diff --git a/_data/devices/Z00D.yml b/_data/devices/Z00D.yml
index 21f2969..e3c44dc 100644
--- a/_data/devices/Z00D.yml
+++ b/_data/devices/Z00D.yml
@@ -7,7 +7,7 @@ cameras:
 carrier:
 channels: [weekly]
 codename: Z00D
-cpu: x86
+cpu: Intel Atom
 cpu_cores: '2+2'
 cpu_freq: 1.6 GHz
 current_branch: 14.1
diff --git a/_data/devices/Z00L.yml b/_data/devices/Z00L.yml
index 389a650..0e30c05 100644
--- a/_data/devices/Z00L.yml
+++ b/_data/devices/Z00L.yml
@@ -7,7 +7,7 @@ cameras:
 carrier:
 channels: [weekly]
 codename: Z00L
-cpu: Cortex A53
+cpu: Cortex-A53
 cpu_cores: 8/4
 cpu_freq: 1.7/1.0 GHz/1.2 GHz
 current_branch: 14.1
diff --git a/_data/devices/Z00T.yml b/_data/devices/Z00T.yml
index 8dec11d..7f6fdd4 100644
--- a/_data/devices/Z00T.yml
+++ b/_data/devices/Z00T.yml
@@ -7,7 +7,7 @@ cameras:
 carrier:
 channels: [weekly]
 codename: Z00T
-cpu: Cortex A53
+cpu: Cortex-A53
 cpu_cores: '8'
 cpu_freq: 1.7/1.0 GHz
 current_branch: 14.1
diff --git a/_data/devices/addison.yml b/_data/devices/addison.yml
index 74f3d69..d6b9611 100644
--- a/_data/devices/addison.yml
+++ b/_data/devices/addison.yml
@@ -7,7 +7,7 @@ cameras:
 carrier:
 channels: [discontinued]
 codename: addison
-cpu: Cortex A53
+cpu: Cortex-A53
 cpu_cores: '8'
 cpu_freq: 2.0 GHz
 current_branch: 14.1
diff --git a/_data/devices/angler.yml b/_data/devices/angler.yml
index a5f0f73..e2d4d68 100644
--- a/_data/devices/angler.yml
+++ b/_data/devices/angler.yml
@@ -7,7 +7,7 @@ cameras:
 carrier:
 channels: [weekly]
 codename: angler
-cpu: Quad-core Cortex-A53 & Quad-core Cortex-A57
+cpu: Cortex-A53 & Cortex-A57
 cpu_cores: '8'
 cpu_freq: 1.55 GHz & 2.0 GHz
 current_branch: 14.1
diff --git a/_data/devices/armani.yml b/_data/devices/armani.yml
index befb6fa..c722b8d 100644
--- a/_data/devices/armani.yml
+++ b/_data/devices/armani.yml
@@ -7,7 +7,7 @@ cameras:
 carrier:
 channels: [weekly]
 codename: armani
-cpu: Cortex A7
+cpu: Cortex-A7
 cpu_cores: '4'
 cpu_freq: 1.6 GHz
 current_branch: 14.1
diff --git a/_data/devices/athene.yml b/_data/devices/athene.yml
index d3da028..806435f 100644
--- a/_data/devices/athene.yml
+++ b/_data/devices/athene.yml
@@ -7,7 +7,7 @@ cameras:
 carrier:
 channels: [weekly]
 codename: athene
-cpu: Cortex A53
+cpu: Cortex-A53
 cpu_cores: '8'
 cpu_freq: 1.5 GHz
 current_branch: 14.1
diff --git a/_data/devices/chagallwifi.yml b/_data/devices/chagallwifi.yml
index d25e1f2..76e1096 100644
--- a/_data/devices/chagallwifi.yml
+++ b/_data/devices/chagallwifi.yml
@@ -7,7 +7,7 @@ cameras:
 carrier:
 channels: [weekly]
 codename: chagallwifi
-cpu: Cortex A15 & A7
+cpu: Cortex-A15 & Cortex-A7
 cpu_cores: '8'
 cpu_freq: 4 x 1.9 GHz & 4 x 1.3 GHz
 current_branch: 14.1
diff --git a/_data/devices/che10.yml b/_data/devices/che10.yml
index e5b51c6..2fea0f4 100644
--- a/_data/devices/che10.yml
+++ b/_data/devices/che10.yml
@@ -7,7 +7,7 @@ cameras:
 carrier:
 channels: [weekly]
 codename: che10
-cpu: Cortex A53
+cpu: Cortex-A53
 cpu_cores: '4'
 cpu_freq: 1.2 GHz
 current_branch: 14.1
diff --git a/_data/devices/cherry.yml b/_data/devices/cherry.yml
index 047b985..6be7fea 100644
--- a/_data/devices/cherry.yml
+++ b/_data/devices/cherry.yml
@@ -7,7 +7,7 @@ cameras:
 carrier:
 channels: [weekly]
 codename: cherry
-cpu: Cortex A53
+cpu: Cortex-A53
 cpu_cores: '4'
 cpu_freq: 1.2 GHz
 current_branch: 14.1
diff --git a/_data/devices/condor.yml b/_data/devices/condor.yml
index c8192dd..fab53a8 100644
--- a/_data/devices/condor.yml
+++ b/_data/devices/condor.yml
@@ -6,7 +6,7 @@ cameras:
 carrier:
 channels: [weekly]
 codename: condor
-cpu: Cortex A7
+cpu: Cortex-A7
 cpu_cores: '2'
 cpu_freq: 1.2 GHz
 current_branch: 14.1
diff --git a/_data/devices/crackling.yml b/_data/devices/crackling.yml
index 96f10c0..8eef00c 100644
--- a/_data/devices/crackling.yml
+++ b/_data/devices/crackling.yml
@@ -7,7 +7,7 @@ cameras:
 carrier:
 channels: [weekly]
 codename: crackling
-cpu: 64-bit Cortex A53
+cpu: Cortex-A53
 cpu_cores: '4'
 cpu_freq: 1.2 GHz
 current_branch: 14.1
diff --git a/_data/devices/d800.yml b/_data/devices/d800.yml
index b79573e..3b7df81 100644
--- a/_data/devices/d800.yml
+++ b/_data/devices/d800.yml
@@ -7,7 +7,7 @@ cameras:
 carrier: AT&T
 channels: [weekly]
 codename: d800
-cpu: Qualcomm Krait 400
+cpu: Krait 400
 cpu_cores: '4'
 cpu_freq: 2.26 GHz
 current_branch: 14.1
diff --git a/_data/devices/d801.yml b/_data/devices/d801.yml
index 7e56682..46ae9de 100644
--- a/_data/devices/d801.yml
+++ b/_data/devices/d801.yml
@@ -7,7 +7,7 @@ cameras:
 carrier: T-Mobile
 channels: [weekly]
 codename: d801
-cpu: Qualcomm Krait 400
+cpu: Krait 400
 cpu_cores: '4'
 cpu_freq: 2.26 GHz
 current_branch: 14.1
diff --git a/_data/devices/d802.yml b/_data/devices/d802.yml
index 1ab00c8..f4115df 100644
--- a/_data/devices/d802.yml
+++ b/_data/devices/d802.yml
@@ -7,7 +7,7 @@ cameras:
 carrier:
 channels: [weekly]
 codename: d802
-cpu: Qualcomm Krait 400
+cpu: Krait 400
 cpu_cores: '4'
 cpu_freq: 2.26 GHz
 current_branch: 14.1
diff --git a/_data/devices/d803.yml b/_data/devices/d803.yml
index dc4b17d..d5fa7ac 100644
--- a/_data/devices/d803.yml
+++ b/_data/devices/d803.yml
@@ -7,7 +7,7 @@ cameras:
 carrier:
 channels: [weekly]
 codename: d803
-cpu: Qualcomm Krait 400
+cpu: Krait 400
 cpu_cores: '4'
 cpu_freq: 2.26 GHz
 current_branch: 14.1
diff --git a/_data/devices/espresso3g.yml b/_data/devices/espresso3g.yml
index ee70f3b..52199a9 100644
--- a/_data/devices/espresso3g.yml
+++ b/_data/devices/espresso3g.yml
@@ -7,7 +7,7 @@ cameras:
 carrier:
 channels: [weekly]
 codename: espresso3g
-cpu: OMAP4
+cpu: Cortex-A9
 cpu_cores: '2'
 cpu_freq: 1 GHz
 current_branch: 13.0
diff --git a/_data/devices/espressowifi.yml b/_data/devices/espressowifi.yml
index dd18d5d..42b7392 100644
--- a/_data/devices/espressowifi.yml
+++ b/_data/devices/espressowifi.yml
@@ -7,7 +7,7 @@ cameras:
 carrier:
 channels: [weekly]
 codename: espressowifi
-cpu: OMAP4
+cpu: Cortex-A9
 cpu_cores: '2'
 cpu_freq: 1 GHz
 current_branch: 13.0
diff --git a/_data/devices/falcon.yml b/_data/devices/falcon.yml
index ddd0d88..5df99d5 100644
--- a/_data/devices/falcon.yml
+++ b/_data/devices/falcon.yml
@@ -7,7 +7,7 @@ cameras:
 carrier: AT&T,TMobile,Boost, International GSM
 channels: [weekly]
 codename: falcon
-cpu: Cortex A7
+cpu: Cortex-A7
 cpu_cores: '4'
 cpu_freq: 1.2 GHz
 current_branch: 14.1
diff --git a/_data/devices/flounder.yml b/_data/devices/flounder.yml
index 2f9819a..abecc13 100644
--- a/_data/devices/flounder.yml
+++ b/_data/devices/flounder.yml
@@ -7,7 +7,7 @@ cameras:
 carrier:
 channels: [weekly]
 codename: flounder
-cpu: Denver (ARMv8 64-bit)
+cpu: Denver
 cpu_cores: '2'
 cpu_freq: 2.3 GHz
 current_branch: 14.1
diff --git a/_data/devices/flounder_lte.yml b/_data/devices/flounder_lte.yml
index 64427a1..d440d01 100644
--- a/_data/devices/flounder_lte.yml
+++ b/_data/devices/flounder_lte.yml
@@ -7,7 +7,7 @@ cameras:
 carrier:
 channels: [weekly]
 codename: flounder_lte
-cpu: Denver (ARMv8 64-bit)
+cpu: Denver
 cpu_cores: '2'
 cpu_freq: 2.3 GHz
 current_branch: 14.1
diff --git a/_data/devices/foster.yml b/_data/devices/foster.yml
index 0d459e8..c849bf6 100644
--- a/_data/devices/foster.yml
+++ b/_data/devices/foster.yml
@@ -5,7 +5,7 @@ cameras: []
 carrier:
 channels: [weekly]
 codename: foster
-cpu: Cortex A57
+cpu: Cortex-A57
 cpu_cores: '4'
 cpu_freq: 1.9 GHz
 current_branch: 14.1
diff --git a/_data/devices/g2m.yml b/_data/devices/g2m.yml
index 398212f..921f9c1 100644
--- a/_data/devices/g2m.yml
+++ b/_data/devices/g2m.yml
@@ -7,7 +7,7 @@ cameras:
 carrier:
 channels: [weekly]
 codename: g2m
-cpu: Cortex A7
+cpu: Cortex-A7
 cpu_cores: '4'
 cpu_freq: 1.2 GHz
 current_branch: 14.1
diff --git a/_data/devices/gohan.yml b/_data/devices/gohan.yml
index 956e923..be3854a 100644
--- a/_data/devices/gohan.yml
+++ b/_data/devices/gohan.yml
@@ -7,7 +7,7 @@ cameras:
 carrier:
 channels: [weekly]
 codename: gohan
-cpu: Cortex A72
+cpu: Cortex-A72
 cpu_cores: '8'
 cpu_freq: 1.8 GHz
 current_branch: 14.1
diff --git a/_data/devices/gtelwifiue.yml b/_data/devices/gtelwifiue.yml
index 7699c6c..fc4fb25 100644
--- a/_data/devices/gtelwifiue.yml
+++ b/_data/devices/gtelwifiue.yml
@@ -7,7 +7,7 @@ cameras:
 carrier:
 channels: [weekly]
 codename: gtelwifiue
-cpu: Cortex A53
+cpu: Cortex-A53
 cpu_cores: 4
 cpu_freq: 1.2 GHz
 current_branch: 14.1
diff --git a/_data/devices/gtesqltespr.yml b/_data/devices/gtesqltespr.yml
index c0ae2a5..30929dd 100644
--- a/_data/devices/gtesqltespr.yml
+++ b/_data/devices/gtesqltespr.yml
@@ -7,7 +7,7 @@ cameras:
 carrier: Sprint
 channels: [weekly]
 codename: gtesqltespr
-cpu: Cortex A53
+cpu: Cortex-A53
 cpu_cores: 4
 cpu_freq: 1.2 GHz
 current_branch: 14.1
diff --git a/_data/devices/gts210ltexx.yml b/_data/devices/gts210ltexx.yml
index 663bdf2..1e095ae 100644
--- a/_data/devices/gts210ltexx.yml
+++ b/_data/devices/gts210ltexx.yml
@@ -7,7 +7,7 @@ cameras:
 carrier:
 channels: [weekly]
 codename: gts210ltexx
-cpu: Cortex A57/A53
+cpu: Cortex-A53 & Cortex-A57
 cpu_cores: '8'
 cpu_freq: 4 x 1.9 GHz, 4 x 1.3 GHz
 current_branch: 14.1
diff --git a/_data/devices/gts210vewifi.yml b/_data/devices/gts210vewifi.yml
index 3bec27b..26546e1 100644
--- a/_data/devices/gts210vewifi.yml
+++ b/_data/devices/gts210vewifi.yml
@@ -7,7 +7,7 @@ cameras:
 carrier:
 channels: [weekly]
 codename: gts210vewifi
-cpu: Cortex A53 & A72
+cpu: Cortex-A53 & Cortex-A72
 cpu_cores: '8'
 cpu_freq: 4 x 1.8 GHz & 4 x 1.4 GHz
 current_branch: 14.1
diff --git a/_data/devices/gts210wifi.yml b/_data/devices/gts210wifi.yml
index 58fe553..1c22720 100644
--- a/_data/devices/gts210wifi.yml
+++ b/_data/devices/gts210wifi.yml
@@ -7,7 +7,7 @@ cameras:
 carrier:
 channels: [weekly]
 codename: gts210wifi
-cpu: Cortex A57/A53
+cpu: Cortex-A53 & Cortex-A57
 cpu_cores: '8'
 cpu_freq: 4 x 1.9 GHz, 4 x 1.3 GHz
 current_branch: 14.1
diff --git a/_data/devices/gts28vewifi.yml b/_data/devices/gts28vewifi.yml
index 827d0f5..9bcf285 100644
--- a/_data/devices/gts28vewifi.yml
+++ b/_data/devices/gts28vewifi.yml
@@ -7,7 +7,7 @@ cameras:
 carrier:
 channels: [weekly]
 codename: gts28vewifi
-cpu: Cortex A53 & A72
+cpu: Cortex-A53 & Cortex-A72
 cpu_cores: '8'
 cpu_freq: 4 x 1.8 GHz & 4 x 1.4 GHz
 current_branch: 14.1
diff --git a/_data/devices/ha3g.yml b/_data/devices/ha3g.yml
index 27c54af..a1cd91c 100644
--- a/_data/devices/ha3g.yml
+++ b/_data/devices/ha3g.yml
@@ -7,7 +7,7 @@ cameras:
 carrier:
 channels: [weekly]
 codename: ha3g
-cpu: Cortex A15 & A7
+cpu: Cortex-A15 & Cortex-A7
 cpu_cores: '8'
 cpu_freq: 4 x 1.9 GHz & 4 x 1.3 GHz
 current_branch: 14.1
diff --git a/_data/devices/hammerhead.yml b/_data/devices/hammerhead.yml
index 841d56f..56c6c1f 100644
--- a/_data/devices/hammerhead.yml
+++ b/_data/devices/hammerhead.yml
@@ -7,7 +7,7 @@ cameras:
 carrier:
 channels: [weekly]
 codename: hammerhead
-cpu: Snapdragon 800 Krait 400
+cpu: Krait 400
 cpu_cores: '4'
 cpu_freq: 2.26 GHz
 current_branch: 14.1
diff --git a/_data/devices/harpia.yml b/_data/devices/harpia.yml
index 4590039..dc32d9a 100644
--- a/_data/devices/harpia.yml
+++ b/_data/devices/harpia.yml
@@ -7,7 +7,7 @@ cameras:
 carrier: U.S. Cellular, International GSM
 channels: [weekly]
 codename: harpia
-cpu: Cortex A53
+cpu: Cortex-A53
 cpu_cores: '4'
 cpu_freq: 1.2 GHz
 current_branch: 14.1
diff --git a/_data/devices/hero2lte.yml b/_data/devices/hero2lte.yml
index e038f37..6646617 100644
--- a/_data/devices/hero2lte.yml
+++ b/_data/devices/hero2lte.yml
@@ -7,7 +7,7 @@ cameras:
 carrier:
 channels: [weekly]
 codename: hero2lte
-cpu: M1 + Cortex-A53
+cpu: Exynos M1 & Cortex-A53
 cpu_cores: '8'
 cpu_freq: 4 x 2.3 GHz + 4 x 1.6 GHz
 current_branch: 14.1
diff --git a/_data/devices/herolte.yml b/_data/devices/herolte.yml
index 73c0cdf..bc33f1b 100644
--- a/_data/devices/herolte.yml
+++ b/_data/devices/herolte.yml
@@ -7,7 +7,7 @@ cameras:
 carrier:
 channels: [weekly]
 codename: herolte
-cpu: M1 + Cortex-A53
+cpu: Exynos M1 & Cortex-A53
 cpu_cores: '8'
 cpu_freq: 4 x 2.3 GHz + 4 x 1.6 GHz
 current_branch: 14.1
diff --git a/_data/devices/hiaeuhl.yml b/_data/devices/hiaeuhl.yml
index 4e39a68..c4adfe5 100644
--- a/_data/devices/hiaeuhl.yml
+++ b/_data/devices/hiaeuhl.yml
@@ -7,7 +7,7 @@ cameras:
 carrier:
 channels: [weekly]
 codename: hiaeuhl
-cpu: Cortex A53
+cpu: Cortex-A53
 cpu_cores: '8'
 cpu_freq: 4 x 1.5 GHz & 4 x 1.2 GHz
 current_branch: 14.1
diff --git a/_data/devices/hiaeul.yml b/_data/devices/hiaeul.yml
index 6545ecd..b537bf0 100644
--- a/_data/devices/hiaeul.yml
+++ b/_data/devices/hiaeul.yml
@@ -7,7 +7,7 @@ cameras:
 carrier:
 channels: [weekly]
 codename: hiaeul
-cpu: Cortex A53
+cpu: Cortex-A53
 cpu_cores: '8'
 cpu_freq: 4 x 1.5 GHz & 4 x 1.2 GHz
 current_branch: 14.1
diff --git a/_data/devices/himaul.yml b/_data/devices/himaul.yml
index 2a24b33..f6748a7 100644
--- a/_data/devices/himaul.yml
+++ b/_data/devices/himaul.yml
@@ -7,7 +7,7 @@ cameras:
 carrier:
 channels: [weekly]
 codename: himaul
-cpu: Cortex A53 & A57
+cpu: Cortex-A53 & Cortex-A57
 cpu_cores: '8'
 cpu_freq: 4 x 1.5 Ghz + 4 x 2 Ghz
 current_branch: 14.1
diff --git a/_data/devices/himawl.yml b/_data/devices/himawl.yml
index 05fbf68..a61231d 100644
--- a/_data/devices/himawl.yml
+++ b/_data/devices/himawl.yml
@@ -7,7 +7,7 @@ cameras:
 carrier: Verizon Wireless
 channels: [weekly]
 codename: himawl
-cpu: Cortex A53 & A57
+cpu: Cortex-A53 & Cortex-A57
 cpu_cores: '8'
 cpu_freq: 4 x 1.5 Ghz + 4 x 2 Ghz
 current_branch: 14.1
diff --git a/_data/devices/hydrogen.yml b/_data/devices/hydrogen.yml
index 07d7949..f4306ae 100644
--- a/_data/devices/hydrogen.yml
+++ b/_data/devices/hydrogen.yml
@@ -7,7 +7,7 @@ cameras:
 carrier:
 channels: [weekly]
 codename: hydrogen
-cpu: Cortex A53 & A72
+cpu: Cortex-A53 & Cortex-A72
 cpu_cores: '6'
 cpu_freq: 4 x 1.4 GHz & 2 x 1.8 GHz
 current_branch: 14.1
diff --git a/_data/devices/i9100.yml b/_data/devices/i9100.yml
index c715e1b..796168d 100644
--- a/_data/devices/i9100.yml
+++ b/_data/devices/i9100.yml
@@ -7,7 +7,7 @@ cameras:
 carrier:
 channels: [weekly]
 codename: i9100
-cpu: Cortex A9
+cpu: Cortex-A9
 cpu_cores: '2'
 cpu_freq: 1.2 GHz
 current_branch: 14.1
diff --git a/_data/devices/i9300.yml b/_data/devices/i9300.yml
index 24f1b2c..b249dab 100644
--- a/_data/devices/i9300.yml
+++ b/_data/devices/i9300.yml
@@ -6,7 +6,7 @@ cameras:
 carrier:
 channels: [weekly]
 codename: i9300
-cpu: Cortex A9
+cpu: Cortex-A9
 cpu_cores: 4
 cpu_freq: 1.4GHz
 current_branch: 14.1
diff --git a/_data/devices/i9305.yml b/_data/devices/i9305.yml
index a767699..8e64b98 100644
--- a/_data/devices/i9305.yml
+++ b/_data/devices/i9305.yml
@@ -7,7 +7,7 @@ cameras:
 carrier:
 channels: [discontinued]
 codename: i9305
-cpu: Cortex A9
+cpu: Cortex-A9
 cpu_cores: '4'
 cpu_freq: 1.4 GHz
 current_branch: 14.1
diff --git a/_data/devices/ido.yml b/_data/devices/ido.yml
index 7679862..20e41a4 100644
--- a/_data/devices/ido.yml
+++ b/_data/devices/ido.yml
@@ -7,7 +7,7 @@ cameras:
 carrier:
 channels: [weekly]
 codename: ido
-cpu: Cortex A53 & A53
+cpu: Cortex-A53 & Cortex-A53
 cpu_cores: '8'
 cpu_freq: 4 x 1.5 GHz & 4 x 1.2 GHz
 current_branch: 14.1
diff --git a/_data/devices/ivy.yml b/_data/devices/ivy.yml
index 06fbbcf..14738dc 100644
--- a/_data/devices/ivy.yml
+++ b/_data/devices/ivy.yml
@@ -7,7 +7,7 @@ cameras:
 carrier:
 channels: [weekly]
 codename: ivy
-cpu: Cortex A53 & A57
+cpu: Cortex-A53 & Cortex-A57
 cpu_cores: '8'
 cpu_freq: 4 x 2.0 Ghz + 4 x 1.5 Ghz
 current_branch: 14.1
diff --git a/_data/devices/jag3gds.yml b/_data/devices/jag3gds.yml
index 91d69d8..0177c50 100644
--- a/_data/devices/jag3gds.yml
+++ b/_data/devices/jag3gds.yml
@@ -7,7 +7,7 @@ cameras:
 carrier:
 channels: [weekly]
 codename: jag3gds
-cpu: Cortex A7
+cpu: Cortex-A7
 cpu_cores: '4'
 cpu_freq: 1.2 GHz
 current_branch: 14.1
diff --git a/_data/devices/jagnm.yml b/_data/devices/jagnm.yml
index 2a27e05..ce20376 100644
--- a/_data/devices/jagnm.yml
+++ b/_data/devices/jagnm.yml
@@ -7,7 +7,7 @@ cameras:
 carrier:
 channels: [weekly]
 codename: jagnm
-cpu: Cortex A7
+cpu: Cortex-A7
 cpu_cores: '4'
 cpu_freq: 1.2 GHz
 current_branch: 14.1
diff --git a/_data/devices/jalebi.yml b/_data/devices/jalebi.yml
index b9c1c30..e977228 100644
--- a/_data/devices/jalebi.yml
+++ b/_data/devices/jalebi.yml
@@ -7,7 +7,7 @@ cameras:
 carrier:
 channels: [weekly]
 codename: jalebi
-cpu: Cortex A53
+cpu: Cortex-A53
 cpu_cores: '4'
 cpu_freq: 1.2 Ghz
 current_branch: 14.1
diff --git a/_data/devices/k3gxx.yml b/_data/devices/k3gxx.yml
index 58ead1d..f400d0a 100644
--- a/_data/devices/k3gxx.yml
+++ b/_data/devices/k3gxx.yml
@@ -7,7 +7,7 @@ cameras:
 carrier:
 channels: [weekly]
 codename: k3gxx
-cpu: Cortex A15 & A7
+cpu: Cortex-A15 & Cortex-A7
 cpu_cores: '8'
 cpu_freq: 4 x 1.9 GHz & 4 x 1.3 GHz
 current_branch: 14.1
diff --git a/_data/devices/karin.yml b/_data/devices/karin.yml
index a884419..636d810 100644
--- a/_data/devices/karin.yml
+++ b/_data/devices/karin.yml
@@ -7,7 +7,7 @@ cameras:
 carrier:
 channels: [weekly]
 codename: karin
-cpu: Cortex A53 & A57
+cpu: Cortex-A53 & Cortex-A57
 cpu_cores: '8'
 cpu_freq: 4 x 2.0 Ghz + 4 x 1.5 Ghz
 current_branch: 14.1
diff --git a/_data/devices/karin_windy.yml b/_data/devices/karin_windy.yml
index 64d6a92..63821bc 100644
--- a/_data/devices/karin_windy.yml
+++ b/_data/devices/karin_windy.yml
@@ -7,7 +7,7 @@ cameras:
 carrier:
 channels: [weekly]
 codename: karin_windy
-cpu: Cortex A53 & A57
+cpu: Cortex-A53 & Cortex-A57
 cpu_cores: '8'
 cpu_freq: 4 x 2.0 Ghz + 4 x 1.5 Ghz
 current_branch: 14.1
diff --git a/_data/devices/kenzo.yml b/_data/devices/kenzo.yml
index f1c8472..0e3c1c4 100644
--- a/_data/devices/kenzo.yml
+++ b/_data/devices/kenzo.yml
@@ -7,7 +7,7 @@ cameras:
 carrier:
 channels: [weekly]
 codename: kenzo
-cpu: Cortex A53 & A72
+cpu: Cortex-A53 & Cortex-A72
 cpu_cores: '6'
 cpu_freq: 4 x 1.4 GHz & 2 x 1.8 GHz
 current_branch: 14.1
diff --git a/_data/devices/kipper.yml b/_data/devices/kipper.yml
index d36cd5c..df7f21c 100644
--- a/_data/devices/kipper.yml
+++ b/_data/devices/kipper.yml
@@ -7,7 +7,7 @@ cameras:
 carrier:
 channels: [weekly]
 codename: kipper
-cpu: Cortex A53
+cpu: Cortex-A53
 cpu_cores: '8'
 cpu_freq: 1.5 GHz
 current_branch: 14.1
diff --git a/_data/devices/klimtwifi.yml b/_data/devices/klimtwifi.yml
index b8c505d..7c7ff4c 100644
--- a/_data/devices/klimtwifi.yml
+++ b/_data/devices/klimtwifi.yml
@@ -7,7 +7,7 @@ cameras:
 carrier:
 channels: [weekly]
 codename: klimtwifi
-cpu: Cortex A15 & A7
+cpu: Cortex-A15 & Cortex-A7
 cpu_cores: '8'
 cpu_freq: 4 x 1.9 GHz & 4 x 1.3 GHz
 current_branch: 14.1
diff --git a/_data/devices/lettuce.yml b/_data/devices/lettuce.yml
index a6b79ee..d6e6bef 100644
--- a/_data/devices/lettuce.yml
+++ b/_data/devices/lettuce.yml
@@ -7,7 +7,7 @@ cameras:
 carrier:
 channels: [weekly]
 codename: lettuce
-cpu: Cortex A53
+cpu: Cortex-A53
 cpu_cores: '4'
 cpu_freq: 1.2 GHz
 current_branch: 14.1
diff --git a/_data/devices/lux.yml b/_data/devices/lux.yml
index 4eb7e44..b47d43c 100644
--- a/_data/devices/lux.yml
+++ b/_data/devices/lux.yml
@@ -7,7 +7,7 @@ cameras:
 carrier:
 channels: [weekly]
 codename: lux
-cpu: Cortex A53
+cpu: Cortex-A53
 cpu_cores: '8'
 cpu_freq: 1.7 GHz
 current_branch: 14.1
diff --git a/_data/devices/m216.yml b/_data/devices/m216.yml
index 49b6161..511c609 100644
--- a/_data/devices/m216.yml
+++ b/_data/devices/m216.yml
@@ -7,7 +7,7 @@ cameras:
 carrier:
 channels: [weekly]
 codename: m216
-cpu: Cortex A53
+cpu: Cortex-A53
 cpu_cores: '4'
 cpu_freq: 1.2 GHz
 current_branch: 14.1
diff --git a/_data/devices/maguro.yml b/_data/devices/maguro.yml
index 1cf2b53..5028604 100644
--- a/_data/devices/maguro.yml
+++ b/_data/devices/maguro.yml
@@ -7,7 +7,7 @@ cameras:
 carrier:
 channels: [weekly]
 codename: maguro
-cpu: OMAP4
+cpu: Cortex-A9
 cpu_cores: '2'
 cpu_freq: 1.2 GHz
 current_branch: 13.0
diff --git a/_data/devices/merlin.yml b/_data/devices/merlin.yml
index cb07a41..ffed778 100644
--- a/_data/devices/merlin.yml
+++ b/_data/devices/merlin.yml
@@ -7,7 +7,7 @@ cameras:
 carrier:
 channels: [weekly]
 codename: merlin
-cpu: Cortex A53
+cpu: Cortex-A53
 cpu_cores: '8'
 cpu_freq: 1.7 GHz
 current_branch: 14.1
diff --git a/_data/devices/mido.yml b/_data/devices/mido.yml
index 152f454..25e78d3 100644
--- a/_data/devices/mido.yml
+++ b/_data/devices/mido.yml
@@ -7,7 +7,7 @@ cameras:
 carrier:
 channels: [weekly]
 codename: mido
-cpu: Cortex A53
+cpu: Cortex-A53
 cpu_cores: '8'
 cpu_freq: 2.0 GHz
 current_branch: 14.1
diff --git a/_data/devices/mondrianwifi.yml b/_data/devices/mondrianwifi.yml
index 1c1abcc..60e3e9e 100644
--- a/_data/devices/mondrianwifi.yml
+++ b/_data/devices/mondrianwifi.yml
@@ -7,7 +7,7 @@ cameras:
 carrier:
 channels: [weekly]
 codename: mondrianwifi
-cpu: Snapdragon 800 Krait 400
+cpu: Krait 400
 cpu_cores: '4'
 cpu_freq: 2.26 GHz
 current_branch: 13.0
diff --git a/_data/devices/mt2.yml b/_data/devices/mt2.yml
index e50cff4..afa011d 100644
--- a/_data/devices/mt2.yml
+++ b/_data/devices/mt2.yml
@@ -7,7 +7,7 @@ cameras:
 carrier:
 channels: [discontinued]
 codename: mt2
-cpu: Cortex A7
+cpu: Cortex-A7
 cpu_cores: '4'
 cpu_freq: 1.6 GHz
 current_branch: 14.1
diff --git a/_data/devices/n1awifi.yml b/_data/devices/n1awifi.yml
index 3b2fc87..f7f959b 100644
--- a/_data/devices/n1awifi.yml
+++ b/_data/devices/n1awifi.yml
@@ -7,7 +7,7 @@ cameras:
 carrier:
 channels: [weekly]
 codename: n1awifi
-cpu: Cortex A15 & A7
+cpu: Cortex-A15 & Cortex-A7
 cpu_cores: '8'
 cpu_freq: 4 x 1.9 GHz & 4 x 1.3 GHz
 current_branch: 14.1
diff --git a/_data/devices/n2awifi.yml b/_data/devices/n2awifi.yml
index 0809912..7704c87 100644
--- a/_data/devices/n2awifi.yml
+++ b/_data/devices/n2awifi.yml
@@ -7,7 +7,7 @@ cameras:
 carrier:
 channels: [weekly]
 codename: n2awifi
-cpu: Cortex A15 & A7
+cpu: Cortex-A15 & Cortex-A7
 cpu_cores: '8'
 cpu_freq: 4 x 1.9 GHz & 4 x 1.3 GHz
 current_branch: 14.1
diff --git a/_data/devices/n5100.yml b/_data/devices/n5100.yml
index 24c6d78..2b508d1 100644
--- a/_data/devices/n5100.yml
+++ b/_data/devices/n5100.yml
@@ -7,7 +7,7 @@ cameras:
 carrier:
 channels: [weekly]
 codename: n5100
-cpu: Cortex A9
+cpu: Cortex-A9
 cpu_cores: '4'
 cpu_freq: 1.6 GHz
 current_branch: 14.1
diff --git a/_data/devices/n5110.yml b/_data/devices/n5110.yml
index 50f5407..f22cbba 100644
--- a/_data/devices/n5110.yml
+++ b/_data/devices/n5110.yml
@@ -7,7 +7,7 @@ cameras:
 carrier:
 channels: [weekly]
 codename: n5110
-cpu: Cortex A9
+cpu: Cortex-A9
 cpu_cores: '4'
 cpu_freq: 1.6 GHz
 current_branch: 14.1
diff --git a/_data/devices/n5120.yml b/_data/devices/n5120.yml
index 483c2c5..c93bfad 100644
--- a/_data/devices/n5120.yml
+++ b/_data/devices/n5120.yml
@@ -7,7 +7,7 @@ cameras:
 carrier:
 channels: [weekly]
 codename: n5120
-cpu: Cortex A9
+cpu: Cortex-A9
 cpu_cores: '4'
 cpu_freq: 1.6 GHz
 current_branch: 14.1
diff --git a/_data/devices/nx512j.yml b/_data/devices/nx512j.yml
index d4ddd3a..c0ab313 100644
--- a/_data/devices/nx512j.yml
+++ b/_data/devices/nx512j.yml
@@ -7,7 +7,7 @@ cameras:
 carrier:
 channels: [weekly]
 codename: nx512j
-cpu: Cortex A53
+cpu: Cortex-A53
 cpu_cores: '8'
 cpu_freq: 1.5 GHz
 current_branch: 14.1
diff --git a/_data/devices/oneplus2.yml b/_data/devices/oneplus2.yml
index a588420..2b1628e 100644
--- a/_data/devices/oneplus2.yml
+++ b/_data/devices/oneplus2.yml
@@ -7,7 +7,7 @@ cameras:
 carrier:
 channels: [weekly]
 codename: oneplus2
-cpu: Cortex A53 & A57
+cpu: Cortex-A53 & Cortex-A57
 cpu_cores: '8'
 cpu_freq: 4 x 1.5 Ghz + 4 x 1.8 Ghz
 current_branch: 14.1
diff --git a/_data/devices/osprey.yml b/_data/devices/osprey.yml
index 2eaafb5..72e9d80 100644
--- a/_data/devices/osprey.yml
+++ b/_data/devices/osprey.yml
@@ -7,7 +7,7 @@ cameras:
 carrier: U.S. Cellular, Virgin Mobile, International GSM
 channels: [weekly]
 codename: osprey
-cpu: Cortex A53
+cpu: Cortex-A53
 cpu_cores: '4'
 cpu_freq: 1.36 GHz
 current_branch: 14.1
diff --git a/_data/devices/otus.yml b/_data/devices/otus.yml
index 6e7dab9..ebd37f8 100644
--- a/_data/devices/otus.yml
+++ b/_data/devices/otus.yml
@@ -7,7 +7,7 @@ cameras:
 carrier:
 channels: [weekly]
 codename: otus
-cpu: Cortex A7
+cpu: Cortex-A7
 cpu_cores: '4'
 cpu_freq: 1.2 GHz
 current_branch: 14.1
diff --git a/_data/devices/paella.yml b/_data/devices/paella.yml
index e0f29ca..c4378b2 100644
--- a/_data/devices/paella.yml
+++ b/_data/devices/paella.yml
@@ -7,7 +7,7 @@ cameras:
 carrier:
 channels: [weekly]
 codename: paella
-cpu: Cortex A53
+cpu: Cortex-A53
 cpu_cores: '4'
 cpu_freq: 1.4 GHz
 current_branch: 14.1
diff --git a/_data/devices/peach.yml b/_data/devices/peach.yml
index 296de93..0905d69 100644
--- a/_data/devices/peach.yml
+++ b/_data/devices/peach.yml
@@ -7,7 +7,7 @@ cameras:
 carrier:
 channels: [discontinued]
 codename: peach
-cpu: Cortex A53
+cpu: Cortex-A53
 cpu_cores: '4'
 cpu_freq: 1.2 GHz
 current_branch: 14.1
diff --git a/_data/devices/peregrine.yml b/_data/devices/peregrine.yml
index b1ecce7..c019e33 100644
--- a/_data/devices/peregrine.yml
+++ b/_data/devices/peregrine.yml
@@ -7,7 +7,7 @@ cameras:
 carrier:
 channels: [weekly]
 codename: peregrine
-cpu: Cortex A7
+cpu: Cortex-A7
 cpu_cores: '4'
 cpu_freq: 1.2 GHz
 current_branch: 14.1
diff --git a/_data/devices/piccolo.yml b/_data/devices/piccolo.yml
index 317fee5..20059ae 100644
--- a/_data/devices/piccolo.yml
+++ b/_data/devices/piccolo.yml
@@ -7,7 +7,7 @@ cameras:
 carrier:
 channels: [weekly]
 codename: piccolo
-cpu: Cortex A53
+cpu: Cortex-A53
 cpu_cores: '8'
 cpu_freq: 4 x 1.0 GHz + 4 x 1.5 GHz
 current_branch: 14.1
diff --git a/_data/devices/r5.yml b/_data/devices/r5.yml
index da3aab4..a52bf74 100644
--- a/_data/devices/r5.yml
+++ b/_data/devices/r5.yml
@@ -7,7 +7,7 @@ cameras:
 carrier:
 channels: [weekly]
 codename: r5
-cpu: Cortex A53
+cpu: Cortex-A53
 cpu_cores: '8'
 cpu_freq: 1.5 GHz and 1.0 GHz
 current_branch: 14.1
diff --git a/_data/devices/r7plus.yml b/_data/devices/r7plus.yml
index 51221d9..5f44bbb 100644
--- a/_data/devices/r7plus.yml
+++ b/_data/devices/r7plus.yml
@@ -7,7 +7,7 @@ cameras:
 carrier:
 channels: [weekly]
 codename: r7plus
-cpu: Cortex A53
+cpu: Cortex-A53
 cpu_cores: '8'
 cpu_freq: 1.5 GHz and 1.0 GHz
 current_branch: 14.1
diff --git a/_data/devices/r7sf.yml b/_data/devices/r7sf.yml
index f383447..5969f39 100644
--- a/_data/devices/r7sf.yml
+++ b/_data/devices/r7sf.yml
@@ -7,7 +7,7 @@ cameras:
 carrier:
 channels: [weekly]
 codename: r7sf
-cpu: Cortex A53
+cpu: Cortex-A53
 cpu_cores: '8'
 cpu_freq: 1.5 GHz and 1.0 GHz
 current_branch: 14.1
diff --git a/_data/devices/roth.yml b/_data/devices/roth.yml
index cd87295..1a483ab 100644
--- a/_data/devices/roth.yml
+++ b/_data/devices/roth.yml
@@ -5,7 +5,7 @@ cameras: []
 carrier:
 channels: [weekly]
 codename: roth
-cpu: Cortex A15
+cpu: Cortex-A15
 cpu_cores: '4'
 cpu_freq: 1.9 GHz
 current_branch: 13.0
diff --git a/_data/devices/seed.yml b/_data/devices/seed.yml
index 679c5fe..47d6e8d 100644
--- a/_data/devices/seed.yml
+++ b/_data/devices/seed.yml
@@ -7,7 +7,7 @@ cameras:
 carrier:
 channels: [weekly]
 codename: seed
-cpu: Cortex-A53, 64-bit
+cpu: Cortex-A53
 cpu_cores: '4'
 cpu_freq: 1.2 GHz
 current_branch: 14.1
diff --git a/_data/devices/shieldtablet.yml b/_data/devices/shieldtablet.yml
index be7ff38..c48eb47 100644
--- a/_data/devices/shieldtablet.yml
+++ b/_data/devices/shieldtablet.yml
@@ -7,7 +7,7 @@ cameras:
 carrier: AT&T, T-Mobile
 channels: [weekly]
 codename: shieldtablet
-cpu: Cortex A15
+cpu: Cortex-A15
 cpu_cores: '4'
 cpu_freq: 2.2 GHz
 current_branch: 14.1
diff --git a/_data/devices/surnia.yml b/_data/devices/surnia.yml
index ee9bb56..5347507 100644
--- a/_data/devices/surnia.yml
+++ b/_data/devices/surnia.yml
@@ -7,7 +7,7 @@ cameras:
 carrier:
 channels: [weekly]
 codename: surnia
-cpu: Cortex A53
+cpu: Cortex-A53
 cpu_cores: '4'
 cpu_freq: 1.2 GHz
 current_branch: 14.1
diff --git a/_data/devices/suzuran.yml b/_data/devices/suzuran.yml
index 2a94c75..1a0774a 100644
--- a/_data/devices/suzuran.yml
+++ b/_data/devices/suzuran.yml
@@ -7,7 +7,7 @@ cameras:
 carrier:
 channels: [weekly]
 codename: suzuran
-cpu: Cortex A53 & A57
+cpu: Cortex-A53 & Cortex-A57
 cpu_cores: '8'
 cpu_freq: 4 x 2.0 Ghz + 4 x 1.5 Ghz
 current_branch: 14.1
diff --git a/_data/devices/t0lte.yml b/_data/devices/t0lte.yml
index 5f35972..d343433 100644
--- a/_data/devices/t0lte.yml
+++ b/_data/devices/t0lte.yml
@@ -7,7 +7,7 @@ cameras:
 carrier:
 channels: [weekly]
 codename: t0lte
-cpu: Cortex A9
+cpu: Cortex-A9
 cpu_cores: '4'
 cpu_freq: 1.6 GHz
 current_branch: 14.1
diff --git a/_data/devices/tenshi.yml b/_data/devices/tenshi.yml
index ee0a0c9..dddf302 100644
--- a/_data/devices/tenshi.yml
+++ b/_data/devices/tenshi.yml
@@ -7,7 +7,7 @@ cameras:
 carrier:
 channels: [weekly]
 codename: tenshi
-cpu: Cortex A53
+cpu: Cortex-A53
 cpu_cores: '8'
 cpu_freq: 1.4 GHz
 current_branch: 14.1
diff --git a/_data/devices/thea.yml b/_data/devices/thea.yml
index 107a751..38347aa 100644
--- a/_data/devices/thea.yml
+++ b/_data/devices/thea.yml
@@ -7,7 +7,7 @@ cameras:
 carrier:
 channels: [weekly]
 codename: thea
-cpu: Cortex A7
+cpu: Cortex-A7
 cpu_cores: '4'
 cpu_freq: 1.2 GHz
 current_branch: 14.1
diff --git a/_data/devices/titan.yml b/_data/devices/titan.yml
index a52d7cf..4f2a81b 100644
--- a/_data/devices/titan.yml
+++ b/_data/devices/titan.yml
@@ -7,7 +7,7 @@ cameras:
 carrier:
 channels: [weekly]
 codename: titan
-cpu: Cortex A7
+cpu: Cortex-A7
 cpu_cores: '4'
 cpu_freq: 1.2 GHz
 current_branch: 14.1
diff --git a/_data/devices/tomato.yml b/_data/devices/tomato.yml
index c9780a8..fc8e7a8 100644
--- a/_data/devices/tomato.yml
+++ b/_data/devices/tomato.yml
@@ -7,7 +7,7 @@ cameras:
 carrier:
 channels: [weekly]
 codename: tomato
-cpu: Cortex A53
+cpu: Cortex-A53
 cpu_cores: '8'
 cpu_freq: 1.5 GHz
 current_branch: 14.1
diff --git a/_data/devices/toro.yml b/_data/devices/toro.yml
index 4e3028d..b82829d 100644
--- a/_data/devices/toro.yml
+++ b/_data/devices/toro.yml
@@ -7,7 +7,7 @@ cameras:
 carrier:
 channels: [weekly]
 codename: toro
-cpu: OMAP4
+cpu: Cortex-A9
 cpu_cores: '2'
 cpu_freq: 1.2 GHz
 current_branch: 13.0
diff --git a/_data/devices/toroplus.yml b/_data/devices/toroplus.yml
index 8d8df42..0044140 100644
--- a/_data/devices/toroplus.yml
+++ b/_data/devices/toroplus.yml
@@ -7,7 +7,7 @@ cameras:
 carrier:
 channels: [weekly]
 codename: toroplus
-cpu: OMAP4
+cpu: Cortex-A9
 cpu_cores: '2'
 cpu_freq: 1.2 GHz
 current_branch: 13.0
diff --git a/_data/devices/v1awifi.yml b/_data/devices/v1awifi.yml
index 8abc7a1..28bf081 100644
--- a/_data/devices/v1awifi.yml
+++ b/_data/devices/v1awifi.yml
@@ -7,7 +7,7 @@ cameras:
 carrier:
 channels: [weekly]
 codename: v1awifi
-cpu: Cortex A15 & A7
+cpu: Cortex-A15 & Cortex-A7
 cpu_cores: '8'
 cpu_freq: 4 x 1.9 GHz & 4 x 1.3 GHz
 current_branch: 14.1
diff --git a/_data/devices/v400.yml b/_data/devices/v400.yml
index 9c41cb9..80d96c3 100644
--- a/_data/devices/v400.yml
+++ b/_data/devices/v400.yml
@@ -7,7 +7,7 @@ cameras:
 carrier:
 channels: [weekly]
 codename: v400
-cpu: Cortex A7
+cpu: Cortex-A7
 cpu_cores: '4'
 cpu_freq: 1.2 GHz
 current_branch: 14.1
diff --git a/_data/devices/v410.yml b/_data/devices/v410.yml
index 3cacda5..95449a6 100644
--- a/_data/devices/v410.yml
+++ b/_data/devices/v410.yml
@@ -7,7 +7,7 @@ cameras:
 carrier:
 channels: [weekly]
 codename: v410
-cpu: Cortex A7
+cpu: Cortex-A7
 cpu_cores: '4'
 cpu_freq: 1.2 GHz
 current_branch: 14.1
diff --git a/_data/devices/v480.yml b/_data/devices/v480.yml
index 03a50b0..2a20628 100644
--- a/_data/devices/v480.yml
+++ b/_data/devices/v480.yml
@@ -6,7 +6,7 @@ cameras:
 - {flash: '', info: '1.3 MP'}
 channels: [weekly]
 codename: v480
-cpu: Cortex A7
+cpu: Cortex-A7
 cpu_cores: '4'
 cpu_freq: 1.2 GHz
 current_branch: 14.1
diff --git a/_data/devices/v500.yml b/_data/devices/v500.yml
index 601d2ed..661c166 100644
--- a/_data/devices/v500.yml
+++ b/_data/devices/v500.yml
@@ -7,7 +7,7 @@ cameras:
 carrier:
 channels: [weekly]
 codename: v500
-cpu: Qualcomm Krait 300
+cpu: Krait 300
 cpu_cores: '4'
 cpu_freq: 1.7 GHz
 current_branch: 14.1
diff --git a/_data/devices/v521.yml b/_data/devices/v521.yml
index 698ca6b..e1c819b 100644
--- a/_data/devices/v521.yml
+++ b/_data/devices/v521.yml
@@ -7,7 +7,7 @@ cameras:
 carrier: T-Mobile
 channels: [weekly]
 codename: v521
-cpu: Cortex A53
+cpu: Cortex-A53
 cpu_cores: '8'
 cpu_freq: 4 x 1.5 GHz + 4 x 1.2 GHz
 current_branch: 14.1
diff --git a/_data/devices/vegetalte.yml b/_data/devices/vegetalte.yml
index cbc8c05..63ede88 100644
--- a/_data/devices/vegetalte.yml
+++ b/_data/devices/vegetalte.yml
@@ -7,7 +7,7 @@ cameras:
 carrier:
 channels: [weekly]
 codename: vegetalte
-cpu: Cortex A53
+cpu: Cortex-A53
 cpu_cores: '4'
 cpu_freq: 1.2 GHz (E5 4G) / 1.4 GHz (E5s)
 current_branch: 14.1
diff --git a/_data/devices/w5.yml b/_data/devices/w5.yml
index 3b54a60..150d3b6 100644
--- a/_data/devices/w5.yml
+++ b/_data/devices/w5.yml
@@ -7,7 +7,7 @@ cameras:
 carrier:
 channels: [weekly]
 codename: w5
-cpu: Cortex A7
+cpu: Cortex-A7
 cpu_cores: '2'
 cpu_freq: 1.2 GHz
 current_branch: 14.1
diff --git a/_data/devices/w7.yml b/_data/devices/w7.yml
index 2f5b638..4606f84 100644
--- a/_data/devices/w7.yml
+++ b/_data/devices/w7.yml
@@ -7,7 +7,7 @@ cameras:
 carrier:
 channels: [weekly]
 codename: w7
-cpu: Cortex A7
+cpu: Cortex-A7
 cpu_cores: '4'
 cpu_freq: 1.2 GHz
 current_branch: 14.1
diff --git a/_data/devices/wt88047.yml b/_data/devices/wt88047.yml
index f17e5c5..4eaa16f 100644
--- a/_data/devices/wt88047.yml
+++ b/_data/devices/wt88047.yml
@@ -7,7 +7,7 @@ cameras:
 carrier:
 channels: [discontinued]
 codename: wt88047
-cpu: Cortex A53
+cpu: Cortex-A53
 cpu_cores: '4'
 cpu_freq: 1.2 GHz
 current_branch: 14.1
diff --git a/test/schema-06.yml b/test/schema-06.yml
index a27a904..28cbecc 100644
--- a/test/schema-06.yml
+++ b/test/schema-06.yml
@@ -76,6 +76,28 @@ properties:
     "$id": "/properties/codename"
     minLength: 1
     type: string
+  cpu:
+    "$id": "/properties/cpu"
+    enum:
+    - Cortex-A7
+    - Cortex-A9
+    - Cortex-A15
+    - Cortex-A15 & Cortex-A7
+    - Cortex-A53
+    - Cortex-A53 & Cortex-A53
+    - Cortex-A53 & Cortex-A57
+    - Cortex-A53 & Cortex-A72
+    - Cortex-A57
+    - Cortex-A72
+    - Denver
+    - Exynos M1 & Cortex-A53
+    - Intel Atom
+    - Krait
+    - Krait 200
+    - Krait 300
+    - Krait 400
+    - Krait 450
+    - Kryo
   cpu_cores:
     "$id": "/properties/cpu_cores"
     minimum: 1
@@ -222,6 +244,7 @@ required:
 - bluetooth
 - channels
 - codename
+- cpu
 - cpu_cores
 - cpu_freq
 - current_branch
-- 
2.17.1

